### PR TITLE
use simulator in messenger/diagnostic/multiphysics_integrator tests

### PR DESCRIPTION
**Pull request by Dekken**
moved /pharein to /phare/pharein

I would have liked phare/in but "in" is a keyword in python

### CODE REVIEW

File: tests/simulator/per_test.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        input->read();
+    }
+    static void set(PHARE::initializer::PHAREDict& _dict)
+    {
+        auto& dict = PHARE::initializer::PHAREDictHandler::INSTANCE().dict();
+        // hax - migrate to pure python *eventually*
+        if (dict.contains("simulation"))
+        {
+            dict["simulation"]["ions"]       = _dict["ions"];
```
Comment (nicolasaunai) [inline tests/simulator/per_test.h]: if this is not used then maybe remove it?

File: src/simulator/simulator.h [hunk 2]
```diff
@@ -359,7 +311,10 @@ class ISimulator
 template<std::size_t dimension, std::size_t interp_order>
 class Simulator : public ISimulator
 {
-private:
+public:
+    static constexpr size_t dimension_    = dimension;
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/simulator/simulator.h]: it had to be different than the class template but ok

File: tests/amr/messengers/test_messengers.cpp [hunk 3]
```diff
@@ -474,33 +367,27 @@ TEST_F(AfullHybridBasicHierarchy, initializesFieldsOnRefinedLevels)
...
 
 
-TEST_F(AfullHybridBasicHierarchy, initializesParticlesOnRefinedLevels)
+TYPED_TEST(SimulatorTest, initializesParticlesOnRefinedLevels)
 {
-    auto& hierarchy = basicHierarchy->getHierarchy();
-    auto& rm        = hybridModel->resourcesManager;
+    TypeParam sim;
+    auto& hierarchy   = *sim.getHierarchy();
+    auto& hybridModel = *sim.getHybridModel();
```
[the remark this answers is not in the document]
Comment (Dekken) [reply tests/amr/messengers/test_messengers.cpp]: this could be a bit tricky but ok

File: tests/simulator/per_test.h (hunk 1 above)
Comment (Dekken) [reply tests/simulator/per_test.h]: done

File: src/initializer/__init__.py.in [hunk 4]
```diff
@@ -1,15 +1,11 @@
...
-import sys
+import os, sys, subprocess
 
-
-cmake_build_dir = "@CMAKE_BINARY_DIR@"
-cmake_source_dir = "@CMAKE_SOURCE_DIR@"
-pharein_dir      = os.path.join(cmake_source_dir, "subprojects")
 venv_path = os.environ.get('VIRTUAL_ENV')
 
+print("INIT INIT")
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/initializer/__init__.py.in]: it's commented out in my version :\

File: tests/amr/messengers/test_messengers.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dekken) [reply tests/amr/messengers/test_messengers.cpp]: you probably won't like what I've done to make things still work!

File: phare/pharein/maxwellian_fluid_model.py [hunk 5]
```diff
@@ -3,7 +3,9 @@
 
 class MaxwellianFluidModel(object):
 
-    def defaulter(self, value):
+    def defaulter(self, be, value):
+        if be is not None:
+            return be
         if self.dim == 1:
             return lambda x:value + x*0
```
[the remark this answers is not in the document]
Comment (Dekken) [reply phare/pharein/maxwellian_fluid_model.py]: can we consider doing this properly in another PR?
It's likely to pop in a few places.

**Comment by Dekken** [comment]
replaced by https://github.com/PHAREHUB/PHARE/pull/95

**Pull request closed without merging**